xla/service/gpu/transforms/windowed_einsum_handler_test.cc

```diff
@@ -263,28 +264,23 @@ ENTRY main.12_spmd {
   bool changed;
   TF_ASSERT_OK_AND_ASSIGN(changed, gpu_handler.Run(module.get()));
   EXPECT_TRUE(changed);
-  HloInstruction* ag_loop =
-      FindInstructionByName(module->entry_computation(), "while");
   HloInstruction* inst =
       FindInstructionByName(module->entry_computation(), "dot.7");
   // dot.7 should now consume output of the windowed einsum while loop.
   EXPECT_EQ(inst->operand(0)->opcode(), HloOpcode::kGetTupleElement);
   EXPECT_EQ(inst->operand(0)->tuple_index(), 5);
-  EXPECT_EQ(inst->operand(0)->operand(0), ag_loop);
-
-  // while loop's root should now have a chain of DUS.
-  HloInstruction* ag_while_root = ag_loop->while_body()->root_instruction();
-  EXPECT_THAT(ag_while_root,
-              GmockMatch(m::Tuple(
-                  m::Op(), m::Op(), m::Op(), m::Op(), m::Op(),
-                  m::DynamicUpdateSlice(
-                      m::DynamicUpdateSlice(
-                          m::GetTupleElement(m::Parameter())
-                              .WithPredicate([](const HloInstruction* instr) {
-                                return instr->tuple_index() == 5;
-                              }),
-                          m::Op(), m::Op(), m::Op(), m::Op()),
-                      m::Op(), m::Op(), m::Op(), m::Op()))));
+  const HloInstruction* while_loop = inst->operand(0)->operand(0);
+  EXPECT_EQ(while_loop->opcode(), HloOpcode::kWhile);
+  HloComputation* while_body = while_loop->while_body();
+  int64_t dot_count = 0;
+  for (HloInstruction* ins : while_body->MakeInstructionPostOrder()) {
```

Comment:
You should still check there is a DUS chain, as was done before. This chain should still exist even with the while-loop being unrolled, right? In fact, I think the chain will be much longer this time due to the unrolling.

Also it would be good to check the all-gather is removed and replaced with the DUS chain output of the while-loop. But since this isn't already done, it's fine if this PR doesn't address this.

Reply:
i think the DUS output replacement is checked but allgather's presence is not. i can add the check here